### PR TITLE
Catch & handle zero division, overflow, and function eval errors

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
Address #20 

This adds a few more Exception classes to `calc.py` for catching division by zero, overflow, and some other function evaluation errors.

### CODE REVIEW

File: mitxgraders/helpers/calc.py [hunk 1]
```diff
@@ -64,13 +64,50 @@ class FactorialError(CalcError):
...
+class FunctionEvalError(CalcError):
+    """
+    Indicates that something has gone wrong during function evaluation.
+    """
+
+
 class UnableToParse(CalcError):
     """
-    Indicate when a function cannot be parsed
+    Indicate when a expression cannot be parsed
```
Comment (jolyonb) [inline mitxgraders/helpers/calc.py]: Nit: "an expression"

File: mitxgraders/helpers/calc.py [hunk 2]
```diff
@@ -289,32 +326,57 @@ def eval_number(parse_result):
...
+            msg = "There was an error evaluating {name}(...). (Numerical overflow).".format(name=name)
+            raise CalcOverflowError(msg)
+        except Exception as err: # pylint: disable=W0703
+            if isinstance(err, ValueError) and 'factorial' in err.message:
+                # This is thrown when fact() or factorial() is used
+                # that tests on negative and/or non-integer inputs
+                # err.message will be: `factorial() only accepts integral values` or
+                # `factorial() not defined for negative values`
+                raise FactorialError("Error evaluating factorial() or fact() in input. " +
+                                     "These functions can only be used on positive integers.")
```
Comment (jolyonb) [inline mitxgraders/helpers/calc.py]: Nit: fact(0) = 1 (or it should). Change positive to non-negative?

**Comment by jolyonb** [comment]
Two tiny nits. Otherwise, looks good to me!

I've just realized that I really like seeing that all tests are passing using Travis - I don't have to pull the branch manually and run the tests!

**Comment by ChristopherChudzicki** [comment]
Should be good to merge now, then. Yes, the tests are very nice.

**Pull request merged**